pkg/reconciler/v1alpha1/service/service_test.go

```diff
@@ -19,6 +19,7 @@ package service
 import (
 	"fmt"
```

Comment:
What about tests for service.go to make sure it is actually reporting metrics to the mock reporter. See: 
https://github.com/knative/pkg/blob/1f624b77dd9358af2154fe1cd1e5c03cf8985908/controller/controller_test.go#L612 and https://github.com/knative/pkg/blob/master/controller/testing/fake_stats_reporter.go.

Reply:
I'm not sure how to achieve this with TableTest. Are you ok with leaving this for a follow up? Otherwise, I'd appreciate some additional pointers.

Reply:
I have the idea now. However, it involves changing the constructors for all the reconcilers' tests. Should a TODO be enough?

Reply:
Done